piqueserver/player.py

```diff
@@ -265,8 +266,38 @@ def on_team_join(self, team: 'FeatureTeam') -> HookValue:
                     return False
                 self.send_chat('Team is full, moved to %s' % other_team.name)
                 return other_team
+
+        # Add the user to their relevant groups
+        self.user_types.add("all")
+
+        team_usertype = None
+
+        if team is self.protocol.team_1:
+            team_usertype = "team_1"
+        elif team is self.protocol.team_2:
+            team_usertype = "team_2"
+        elif team.spectator:
+            team_usertype = "spectator"
+
+        if team_usertype:
```

Comment:
If no `team_usertype`, shouldn't all the special usertypes still be removed?

Reply:
That's true. team_usertype will never be None in the forseeable future, so the `if` is pretty pointless anyway.